### PR TITLE
fix: go mod tidy and gofmt -s

### PR DESCRIPTION
**Pull request by mvdan**
This makes Go 1.12's mod tidy happy.

And gofmt -s can remove lots of unnecessary redundant types, helping
readability.

### CODE REVIEW

**Comment by mvdan** [comment]
Hmm, I didn't request the reviews. Did this happen automatically?

**Comment by richardlt** [comment]
Yes Github automatically adds code owners from .github folder for review.
Thanks for the PR, I think mgo is not something we use in CDS, maybe this lib can be removed.

**Comment by mvdan** [comment]
mgo is just an indirect module requirement; the change to `go.mod` is just needed so that further calls to `go mod tidy` don't have to figure out the correct mgo version again.

File: go.sum [hunk 1]
```diff
@@ -626,5 +626,9 @@ k8s.io/client-go v10.0.0+incompatible h1:F1IqCqw7oMBzDkqlcBymRq1450wD0eNqLE9jzUr
 k8s.io/client-go v10.0.0+incompatible/go.mod h1:7vJpHMYJwNQCWgzmNV+VYUl1zCObLyodBc8nIyt8L5s=
 k8s.io/klog v0.2.0 h1:0ElL0OHzF3N+OhoJTL0uca20SxtYt4X4+bzHeqrB83c=
 k8s.io/klog v0.2.0/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
+labix.org/v2/mgo v0.0.0-20140701140051-000000000287 h1:L0cnkNl4TfAXzvdrqsYEmxOHOCv2p5I3taaReO8BWFs=
```
Comment (yesnault) [inline go.sum]: Do you know why labix.org/v2/mgo is added here?
If no, can you remove go.mod & go.sum from this PR? 
Thanks a lot

Comment (mvdan) [reply go.sum]: This is what `go mod tidy` required to be happy. They're indirect module requirements. You can use `go mod why -m` for this - for example:

```
$ go mod why -m labix.org/v2/mgo
# labix.org/v2/mgo
github.com/ovh/cds/engine/api/secret
github.com/hashicorp/vault/api
github.com/hashicorp/vault/api.test
github.com/hashicorp/vault/vault
github.com/hashicorp/vault/physical
github.com/hashicorp/consul/lib
github.com/hashicorp/serf/serf
github.com/hashicorp/go-msgpack/codec
github.com/hashicorp/go-msgpack/codec.test
labix.org/v2/mgo/bson
```

**Comment by mvdan** [comment]
I've rebased and re-run both commands.

File: go.sum (hunk 1 above)
Comment (yesnault) [reply go.sum]: hum yes, go mod is so strange :-)

on master branch:
```bash
➜  cds git:(master) ✗ go mod why -m labix.org/v2/mgo
# labix.org/v2/mgo
(main module does not need module labix.org/v2/mgo)
➜  cds git:(master) ✗ go mod why -m gopkg.in/mgo.v2
# gopkg.in/mgo.v2
github.com/ovh/cds/engine/api/secret
github.com/hashicorp/vault/api
github.com/hashicorp/vault/api.test
github.com/hashicorp/vault/vault
github.com/hashicorp/vault/physical
github.com/hashicorp/vault/physical.test
gopkg.in/ory-am/dockertest.v2
gopkg.in/ory-am/dockertest.v2.test
gopkg.in/mgo.v2
```

On your branch, if I put go.mod and go.sum from master branch, then run go mod tidy, I have:

```diff
diff --git a/go.mod b/go.mod
index 349dabdbf..7ba0405df 100644
--- a/go.mod
+++ b/go.mod
@@ -279,8 +279,6 @@ require (
        k8s.io/apimachinery v0.0.0-20190223094358-dcb391cde5ca
        k8s.io/client-go v10.0.0+incompatible
        k8s.io/klog v0.2.0 // indirect
-       labix.org/v2/mgo v0.0.0-20140701140051-000000000287 // indirect
-       launchpad.net/gocheck v0.0.0-20140225173054-000000000087 // indirect
        sigs.k8s.io/yaml v1.1.0 // indirect

 )
diff --git a/go.sum b/go.sum
index 4b168f39e..6c812bbda 100644
--- a/go.sum
+++ b/go.sum
@@ -634,9 +634,5 @@ k8s.io/client-go v10.0.0+incompatible h1:F1IqCqw7oMBzDkqlcBymRq1450wD0eNqLE9jzUr
 k8s.io/client-go v10.0.0+incompatible/go.mod h1:7vJpHMYJwNQCWgzmNV+VYUl1zCObLyodBc8nIyt8L5s=
 k8s.io/klog v0.2.0 h1:0ElL0OHzF3N+OhoJTL0uca20SxtYt4X4+bzHeqrB83c=
 k8s.io/klog v0.2.0/go.mod h1:Gq+BEi5rUBO/HRz0bTSXDUcqjScdoY3a9IHpCEIOOfk=
-labix.org/v2/mgo v0.0.0-20140701140051-000000000287 h1:L0cnkNl4TfAXzvdrqsYEmxOHOCv2p5I3taaReO8BWFs=
-labix.org/v2/mgo v0.0.0-20140701140051-000000000287/go.mod h1:Lg7AYkt1uXJoR9oeSZ3W/8IXLdvOfIITgZnommstyz4=
-launchpad.net/gocheck v0.0.0-20140225173054-000000000087 h1:Izowp2XBH6Ya6rv+hqbceQyw/gSGoXfH/UPoTGduL54=
-launchpad.net/gocheck v0.0.0-20140225173054-000000000087/go.mod h1:hj7XX3B/0A+80Vse0e+BUHsHMTEhd0O4cpUHr/e/BUM=
 sigs.k8s.io/yaml v1.1.0 h1:4A07+ZFc2wgJwo8YNlQpr1rVlgUDlxXHhPJciaPY5gs=
 sigs.k8s.io/yaml v1.1.0/go.mod h1:UJmg0vDUVViEyp3mgSv9WPwZCDxu4rQW1olrI1uml+o=
```

BTW, when we update go.mod & go.sum and as there is some vendor, we have to run 
```bash
make mod
```
to run mod tidy & mod vendor + some clean.

I'll merge your branch on a dev then run make mod to generate vendor.
Thank a lot for your contribution!

**Pull request merged**